### PR TITLE
Modernize project

### LINKED ISSUE #84: Make it work with androidx
This needs to get up to speed so we can use the new androidx framework... androidx.test.espresso.intent.rule

### PR DESCRIPTION
**Pull request by felipecsl**
* Update to latest AGP version
* Upgrade Gradle version
* Migrate to AndroidX

Fixes #84

### CODE REVIEW

File: okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package okreplay.sample
 
 import android.os.Bundle
-import android.support.design.widget.BottomNavigationView
-import android.support.v7.app.AppCompatActivity
+import com.google.android.material.bottomnavigation.BottomNavigationView
```
[the remark this answers is not in the document]
Comment (felipecsl) [reply okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt]: I used the android studio "Migrate to AndroidX" tool so it automatically fixed all imports, except for this one. I'd assume not but haven't checked. I'll double check!

[the remark this answers is not in the document]
Comment (elihart) [reply okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt]: that tool doesn't actually work that great - lmk what you find. if this is good i'll merge it

[the remark this answers is not in the document]
Comment (rossbacher) [reply okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt]: Let me run my script... :-)

[the remark this answers is not in the document]
Comment (rossbacher) [reply okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt]: Actually this is the correct mapping. A lot of the design things went into the material design lib.

**Comment by rossbacher** [comment]
The major changes were the update of the `espresso` and `espresso rules` dependency to `androidx`. But I also cleaned it up a bit and made it more consistent.

**Comment by felipecsl** [comment]
thanks @rossbacher lgtm on green CI

**Comment by rossbacher** [comment]
Also it looks like as if CI is not building the sample app, which is why it passed before (even though appcompat lib was wrong)

**Comment by tmtrademarked** [comment]
This is great to see! A slightly tangential question here: is it possible to get a timeline on a 1.5 release? Our project is using the 1.5 snapshot since we want some of the features that have gone in since 1.4, but that resulted in this change unexpectedly breaking us. Totally my fault for depending on a SNAPSHOT version in a production app, but I'd love to be able to stop having that dangling risk laying around.

File: okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (slott) [reply okreplay-sample/src/main/kotlin/okreplay/sample/MainActivity.kt]: Yeah - auto migrate isn't that great :(
Spend most of a day to clean up after it...

**Pull request merged**